master/internal/db/postgres_experiments.go

```diff
@@ -252,3 +260,126 @@ ORDER BY v.end_time;`, &rows, metricName, experimentID, batchesProcessed, startT
 
 	return trials, endTime, nil
 }
+
+// TopTrialsByMetric chooses the subset of trials from an experiment that recorded the best values
+// for the specified metric at any point during the trial.
+func (db *PgDB) TopTrialsByMetric(experimentID int, maxTrials int, metric string,
+	smallerIsBetter bool) (trials []int32, err error) {
+	order := desc
+	aggregate := max
+	if smallerIsBetter {
+		order = asc
+		aggregate = min
+	}
+	err = db.sql.Select(&trials, fmt.Sprintf(`
+SELECT t.id FROM (
+  SELECT t.id,
+    %s((v.metrics->'validation_metrics'->$1)::text::numeric) as best_metric
```

Comment:
```suggestion
    %s((v.metrics->'validation_metrics'->>$1)::numeric) as best_metric
```
I think this should work too. Non-blocking, don't think it makes much of a difference.

Reply:
IIRC ::text forced it to be treated as a scalar in a way ::numeric didn't, and THEN ::numeric could work. But I'll double check.

Reply:
Yeah, the `->>` should make it text already, instead of jsonb.

Reply:
Confirmed - ::numeric by itself is an invalid conversion from jsonb. ::text flattens it to a scalar that you can then interpret as numeric.

Reply:
But `->>` should return text where `->` returns jsonb is all I was saying. Is that not the caes? Either way it's basically the same.